### ISSUE TITLE
Fixed set of URLs for Proxy.pac

### ISSUE THREAD

**Issue by maxmeyer** [net1957/ProxyCheck#4]
Having an empty PC_PROXY_PAC_URLS would generate an empty input field. Having a predefined PC_PROXY_PAC_URLS variable should generate a drop down field. 

Possible format for `PC_PROXY_PAC_URLS`

~~~
PC_PROXY_PAC_URLS=Desc1,http://example.com/proxy1.pac;Desc2,http://example.com/proxy2.pac
~~~

**Comment by maxmeyer** [comment]
Result with at least one predefined URL:

![unbenannt](https://user-images.githubusercontent.com/114236/31997014-ccc48f70-b98a-11e7-96ce-ac1bbfecfef5.PNG)

**Comment by maxmeyer** [comment]
Can we have something like that?

~~~ruby
module ScriptsHelper
  def available_scripts
    return [] if ENV["PC_PROXY_PAC_URLS"].blank?

    ENV["PC_PROXY_PAC_URLS"].split(/;/).map { |l| l.split(/,/) }.map { |l| l[0] = format('%s | %s', l[0], l[1]); l }
  end

  def available_scripts_without_description
    return [] if ENV["PC_PROXY_PAC_URLS"].blank?

    ENV["PC_PROXY_PAC_URLS"].split(/;/).map { |l| l.split(/,/) }.map { |l| l[1] }
  end
end
~~~

And maybe something like this?

~~~erb
<div class="card">
  <div class="card-divider">
    <h5 class="text-center">Client details</h5>
  </div>
  <div class="card-section">
    <%= simple_form_for @script, remote: true, url: scripts_path, html: { id: 'url_form' } do |f| %>
        <div class="form-inputs clearfix">
          <%= f.error_notification %>
<%# ############ NEW %>
          <% if available_scripts.blank? %>
          <%= f.input :name, as: :text, label: 'Proxy.pac content or URL<input id="compress" class="hide button tiny secondary no-margin float-right" data-disable-with="Wait..." value="Compress script" type="submit">'.html_safe, placeholder: 'Script or script url', required: true %>
        <% else %>
          <%= f.input :name, label: 'Proxy.pac content or URL<input id="compress" class="hide button tiny secondary no-margin float-right" data-disable-with="Wait..." value="Compress script" type="submit">'.html_safe, placeholder: 'Script or script url', collection: available_scripts, include_blank: false %>
        <% end %>
<%# ############ NEW %>

          <%= f.input :ip, label: 'IP', placeholder: 'Client IPv4 (default to 127.0.0.1)', required: false %>
          <%= f.input :url, as: :text, label: 'URL(s)<button id="clear" class="hide button tiny secondary no-margin float-right">Clear</button>'.html_safe, placeholder: 'URL(s) to validate', required: false %>
          <%= f.hidden_field :action, required: false %>
          <p class="help-text">* mandatory field</p>
          <div class="form-actions button-group clearfix">
            <%= f.button :submit, 'Run', id: 'url', data: { disable_with: 'Wait...' } %>
          </div>
        </div>
    <% end %>
  </div>
  <div class="results hide">
    <div class="card-divider clearfix">
      <hr>
      <h5 class="text-center">Results<button id="reset" class="button tiny secondary float-right">Clear results</button></h5>
    </div>
    <div class="card-section">
      <table id="result">
        <thead>
        <tr>
          <td>Url</td>
          <td>Proxy used</td>
        </tr>
        </thead>
        <tbody>
        </tbody>
      </table>
    </div>
  </div>
</div>
~~~~

And a check in the controller?

~~~ruby
  def create
    if available_scripts_without_description.present? && !available_scripts_without_description.include?(script_params[:name])
      render json: { error: true }
      return
    end

    @script = Forms::Script.new(script_params)
    script.valid? ? result : render
  end
~~~

**Comment by net1957** [comment]
Thanks for your input.
Should we allow the input of a proxy url if ENV["PC_PROXY_PAC_URLS"] is populated or should we limit to the list ?

**Comment by net1957** [comment]
implemented in version 0.3.0

**Comment by maxmeyer** [comment]
I think, if it's populated it should a fixed set of URLs only, if it's empty show the normal input field
